samples/translate_from_mic.py

```diff
@@ -0,0 +1,177 @@
+
+    def __exit__(self, type=None, value=None, traceback=None):
+        self._audio_stream.stop_stream()
+        self._audio_stream.close()
+        self.closed = True
+        # Signal the generator to terminate so that the client's
+        # streaming_recognize method will not block the process termination.
+        self._buff.put(None)
+        self._audio_interface.terminate()
+
+    def _fill_buffer(self, in_data, frame_count, time_info, status_flags):
+        """Continuously collect data from the audio stream, into the buffer."""
+        self._buff.put(in_data)
+        return None, pyaudio.paContinue
+
+    def exit(self):
+        self.__exit__()
+
+    def generator(self):
+        while not self.closed:
+            # Use a blocking get() to ensure there's at least one chunk of
+            # data, and stop iteration if the chunk is None, indicating the
+            # end of the audio stream.
+            chunk = self._buff.get()
+            if chunk is None:
+                return
+            data = [chunk]
+
+            # Now consume whatever other data's still buffered.
+            while True:
+                try:
+                    chunk = self._buff.get(block=False)
+                    if chunk is None:
+                        return
+                    data.append(chunk)
+                except queue.Empty:
+                    break
+
+            yield b''.join(data)
+
+
+def listen_print_loop(responses):
+    """Iterates through server responses and prints them.
+
+    The responses passed is a generator that will block until a response
+    is provided by the server.
+    """
+    translation = ''
+    source = ''
+    for response in responses:
+        # Once the transcription settles, the response contains the
+        # END_OF_SINGLE_UTTERANCE event.
+        if (response.speech_event_type ==
+                SpeechEventType.END_OF_SINGLE_UTTERANCE):
+
+            print(u'\nFinal translation: {0}'.format(translation))
+            print(u'Final recognition result: {0}'.format(source))
+            return 0
+
+        result = response.result
+        translation = result.text_translation_result.translation
+        source = result.recognition_result
+
+        print(u'\nPartial translation: {0}'.format(translation))
+        print(u'Partial recognition result: {0}'.format(source))
+
+
+def do_translation_loop():
+    print('Begin speaking...')
+
+    client = media.SpeechTranslationServiceClient()
+
+    speech_config = media.TranslateSpeechConfig(
+        audio_encoding='linear16',
+        source_language_code='en-US',
+        target_language_code='es-ES')
+
+    config = media.StreamingTranslateSpeechConfig(
+        audio_config=speech_config, single_utterance=True)
+
+    # The first request contains the configuration.
+    # Note that audio_content is explicitly set to None.
+    first_request = media.StreamingTranslateSpeechRequest(
+        streaming_config=config, audio_content=None)
```

Comment:
Any construction of a `StreamingTranslateSpeechRequest` that sets both `streaming_config` and `audio_content` is going to have problems. Either one or the other msut be set, but not both.